pod-operation/src/main.rs

```diff
@@ -1,28 +1,31 @@
-use crate::components::pressure_transducer::PressureTransducer;
 use axum::Server;
 use socketioxide::SocketIo;
 use tracing::{error, info};
 use tracing_subscriber::FmtSubscriber;
+
 mod components;
 mod demo;
 mod state_machine;
+
+use crate::components::pressure_transducer::PressureTransducer;
+use crate::components::signal_light::SignalLight;
 use crate::state_machine::StateMachine;
-use std::{thread, time::Duration};
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
 	tracing::subscriber::set_global_default(FmtSubscriber::default())?;
 
 	let (layer, io) = SocketIo::new_layer();
 
-	thread::sleep(Duration::from_millis(1));
+	let signal_light = SignalLight::new();
+	tokio::spawn(demo::blink(signal_light));
 
 	let pressure_transducer = PressureTransducer::new(0x40);
 	tokio::spawn(demo::read_pressure_transducer(pressure_transducer));
 
-	thread::spawn(move || {
+	tokio::spawn(async {
```

Comment:
Question: Just for clarification, by using `tokio` to spawn a new thread instead of `std::thread` and incorporating that with `tokio::time::interval`, does each thread spend exactly 500 ms on every loop?

Reply:
Sort of: Tokio provides a framework for working with asynchronous threads which won't exactly work with `std::thread`, but even with Tokio, the important part is using an interval instead of waiting for sleep because the interval tracks time even when another task is completing (i.e. `StateMachine::tick`) whereas sleep doesn't wait until the task is done. We can't really say it's "exactly" the interval time, but it will definitely be more consistent and closer than not accounting for the task time.